### PR TITLE
Settings: Remember last used tab & move mixer rows setting to "My Profile"

### PR DESCRIPTION
**Pull request by dcorson-ticino-com**
If Settings window shown at program start, last tab is shown.
last state saved in ini file.
Some redundant code concerning tab change removed.

### CODE REVIEW

File: src/clientsettingsdlg.cpp [hunk 1]
```diff
@@ -652,7 +652,7 @@ CClientSettingsDlg::CClientSettingsDlg ( CClient*         pNCliP,
     QObject::connect ( tabSettings, &QTabWidget::currentChanged,
         this, &CClientSettingsDlg::OnTabChanged );
 
-    tabSettings->setCurrentIndex ( iTabIdx );
+    tabSettings->setCurrentIndex ( pSettings->iSettingsTab ); //pSettings->iSettingsTab
```
Comment (pljones) [inline src/clientsettingsdlg.cpp:655]: I guess the comment was whilst copy/pasting?  It can go, right?

File: src/settings.cpp [hunk 2]
```diff
@@ -533,6 +533,13 @@ if ( GetFlagIniSet ( IniXMLDocument, "client", "defcentservaddr", bValue ) )
         bWindowWasShownConnect = bValue;
     }
 
+    // selected Settings Tab
+    if ( GetNumericIniSet ( IniXMLDocument, "client", "settingstab",
```
Comment (pljones) [inline src/settings.cpp:537]: clang-format will have line width somewhere around the 150 mark, so some lines are wrapped a bit narrow.

File: src/settings.cpp [hunk 3]
```diff
@@ -533,6 +533,13 @@ if ( GetFlagIniSet ( IniXMLDocument, "client", "defcentservaddr", bValue ) )
...
+         0, 3, iValue ) )
```
Comment (hoffie) [inline src/settings.cpp:538]: Really minor and haven't tested it, but if `0, 3` is the inclusive range of permitted values and  there are only three tabs, then I think this should rather be `0, 2` to permit indices 0, 1, and 2?

File: src/clientdlg.cpp [hunk 4]
```diff
@@ -858,6 +858,16 @@ void CClientDlg::OnNumClientsChanged ( int iNewNumClients )
...
 }
 
+void CClientDlg::OnOpenGeneralSettings()
+{
+    ShowGeneralSettings( SETTING_TAB_AUDIONET );
+}
+
+void CClientDlg::OnOpenMusicianProfileDialog()
+{
+    ShowGeneralSettings( SETTING_TAB_USER );
```
Comment (hoffie) [inline src/clientdlg.cpp:868]:
```suggestion
    ShowGeneralSettings ( SETTING_TAB_AUDIONET );
}

void CClientDlg::OnOpenMusicianProfileDialog()
{
    ShowGeneralSettings ( SETTING_TAB_USER );
```

File: src/clientsettingsdlg.cpp (hunk 1 above)
Comment (dcorson-ticino-com) [reply src/clientsettingsdlg.cpp:655]: Oh, yes, of course

**Review by hoffie — DISMISSED** [review]
Looks good to me, thanks for solving this so quickly! :)

Tested:
- First/Second/Third tab are properly restored on restart
- Works properly with empty inifile
- Opening settings still switches correctly to the right tab

**Review by hoffie — APPROVED** [review]
I just realized that this PR also includes the changes to move the mixer rows setting to My Profile. I have updated the PR title to make that clear.
Would be good, if you could also make the commit description clearer in this regard?

Nevertheless, this part works properly as well. :)

File: src/clientdlg.cpp [hunk 5]
```diff
@@ -858,6 +858,16 @@ void CClientDlg::OnNumClientsChanged ( int iNewNumClients )
     SetMyWindowTitle ( iNewNumClients );
 }
 
+void CClientDlg::OnOpenGeneralSettings()
```
Comment (pljones) [inline src/clientdlg.cpp:861]: Sorry to be a pain.  If this is `SETTING_TAB_AUDIONET`, can the slot be called `OnOpenAudioNetSettings`, please, for consistency.  Alternatively, change the name of the defined value.

Comment (pljones) [inline src/clientdlg.cpp:861]: I think the slot name needs to change to match what it does -- `OnOpenGeneralSettings` should be `OnOpenAudioNetSettings`.

File: src/clientdlg.cpp [hunk 6]
```diff
@@ -858,6 +858,16 @@ void CClientDlg::OnNumClientsChanged ( int iNewNumClients )
...
+{
+    ShowGeneralSettings ( SETTING_TAB_AUDIONET );
+}
+
+void CClientDlg::OnOpenMusicianProfileDialog()
```
Comment (pljones) [inline src/clientdlg.cpp:866]: Similarly, now there's no dedicated profile dialog, this should change, at least to end `...Settings`.

File: src/clientdlg.cpp [hunk 7]
```diff
@@ -858,6 +858,16 @@ void CClientDlg::OnNumClientsChanged ( int iNewNumClients )
...
+{
+    ShowGeneralSettings ( SETTING_TAB_USER );
+}
+
```
Comment (pljones) [inline src/clientdlg.cpp:870]: And part of me wants a View->Advanced settings menu option to get to that tab...

**Comment by pljones** [comment]
None of my comments are blocking to get this into 3.8.0.

File: src/clientdlg.cpp (hunk 5 above)
Comment (dcorson-ticino-com) [reply src/clientdlg.cpp:861]: For readability that is a plus.
I have been going on the minimal changes of existing, but that may not be the best way to go.
I will clean this up this afternoon.

**Review by pljones — APPROVED** [review]

**Comment by pljones** [comment]
Approved.  Just a couple of notes.

----

I've not checked the ctrl key sequences for clashes.  I tend to look at something like [this](https://support.microsoft.com/en-us/windows/keyboard-shortcuts-in-apps-139014e7-177b-d1f3-eb2e-7298b2599a34) to get an idea on what's "normal" -- i.e. how hard to avoid things like the clipboard sequences.

----

I'm of the opinion we should start slowly moving towards the functor/lambda style connections.

Rather than:
```
    pViewMenu->addAction ( tr ( "A&dvanced Settings..." ), this,
        SLOT ( OnOpenAdvancedSettings() ), QKeySequence ( Qt::CTRL + Qt::Key_D ) );
```
and needing three methods:
```
void CClientDlg::OnOpenAdvancedSettings()
{
    ShowGeneralSettings ( SETTING_TAB_ADVANCED );
}
```
just do it in line:
```
    pViewMenu->addAction ( tr ( "A&dvanced Settings..." ),
        this, [=]{ ShowGeneralSettings ( SETTING_TAB_ADVANCED ); },
        QKeySequence ( Qt::CTRL + Qt::Key_D ) );
```

**Pull request merged**